app/src/main/java/ch/epfl/sweng/erpa/model/UserProfile.java

```diff
@@ -0,0 +1,49 @@
+package ch.epfl.sweng.erpa.model;
+
+import java.util.Objects;
+
+public class UserProfile {
+
+    private String username;
+    private String accessToken;
+
+    public UserProfile(String username, String password) {
+        this.username = username;
+        this.accessToken = password;
+    }
+
+    public String getUsername() {
+        return username;
+    }
+
+    public String getAccessToken() {
+        return accessToken;
+    }
+
+    public void setUsername(String username) {
+        this.username = username;
+    }
+
+    public void setPassword(String password) {
+        this.accessToken = password;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == null || getClass() != o.getClass()) return false;
+        UserProfile user = (UserProfile) o;
+        return Objects.equals(username, user.username) &&
+                Objects.equals(accessToken, user.accessToken);
+    }
+
+    @Override
+    public int hashCode() {
+
+        return Objects.hash(username, accessToken);
+    }
+
+    @Override
+    public String toString() {
+        return accessToken;
```

Comment:
What? Please implement an actual `toString` with a redacted token

Reply:
That's what I said on monday, I created this class but I didn't use it so I left everything as it was

Reply:
I modified this class for signup but didn't change the toString() method now that you point it out I'll do it too